### DISCUSSION TITLE
[Question]: Frigate HTTP API usage (multiple questions)

### DISCUSSION THREAD

**Discussion by iursevla** [blakeblackshear/frigate#12760] [Ask A Question] [answered]
### What is your question:

Hello guys,

I'm building a tool for intruder-detection which is based on Frigate NVR and used some other data points (mostly Shelly devices to avoid almost all false positives).

Right now, I have some questions:

1. What I'm building needs a way to get the last X seconds of video before an event even started while it is still ongoing. I'm listening to the MQTT feed and when I receive the info that an event happened, I want to grab the data as fast as possible and then send it to a remote location (AWS/GC/etc). The API does not show any endpoint for it. I'm missing any endpoint that could give me this data? An alternative is to have my containers in the same volume and get the data directly but I want to avoid this and use the API instead.
2. What's the best way to deal with this: 
   * from 7 AM to 11 PM the system is just connected storing but not detecting 
   * from 11 PM to 7 AM the system should be detecting car/person/etc
   * In the docs we have [this](https://deploy-preview-11419--frigate-docs.netlify.app/configuration/record#can-i-have-continuous-recordings-but-only-at-certain-times) but it doesn't really explain how. Would it be ideal to have two different configs and basically request the endpoint `/api/config/save?save_option=restart` with the specific config or is there other/better way?
3. Is Frigate using the ONVIF events sent by the cameras in any way shape or form? If it is can Frigate pass them through (e.g., publish to MQTT), or do I need to connect to the camera directly?
4. I have an Intel N100 CPU and I'm using a Coral USB TPU. This is [my current configuration](https://gist.github.com/iursevla/c21e095fcecd21b58be504af7cf6c9c2). Looking at my config, what could I change to make it better for Frigate? Do you happen to know the ideal configuration for export/timelapse_args using preset-vaapi, so that I can export a timelapse with 25x speed and minimum final size?

Btw, I've updated the [Frigate OpenAPI specification with more config related endpoints](https://intruder-detection.github.io/frigate-http-api-typescript/)

**Comment by NickM-27** [comment]
1. It's not clear what you mean by "get the last X seconds"
2. Just use MQTT to toggle detect 
3. Frigate supports onvif for PTZ but that is all that is used for

**Reply by iursevla** [reply]
Updated the question. I meant to say "get the last X seconds of video"

**Reply by NickM-27** [reply]
You can use the api to request any time period where recordings exist

**Reply by iursevla** [reply]
What endpoint(s) would you use @NickM-27 ?

**Reply by NickM-27** [reply]
https://docs.frigate.video/integrations/api#get-vodcamerastartstart-timestampendend-timestampindexm3u8

**Reply by iursevla** [reply]
If I use that endpoint (without the `/index.m3u8`)  [here](https://intruder-detection.github.io/frigate-http-api-typescript/#/Recordings/get_api_vod__camera_name__start__start_timestamp__end__end_timestamp_) I get information about the clips for the last minute for example 

<img width="1048" alt="image" src="https://github.com/user-attachments/assets/92eaa32c-3934-4de0-8faa-c98db4484bf8">

Without `/index.m3u8`

<img width="944" alt="image" src="https://github.com/user-attachments/assets/b58aafb2-d408-41b8-82c7-415416276f01">


With `/index.m3u8`

<img width="960" alt="image" src="https://github.com/user-attachments/assets/dd37a023-bfe3-467c-ad4e-4ec8614ccffc">

I even tried in VLC but it does not work. Im a misunderstanding the way to use the endpoint?

PS: In the newest version, those endpoints (`/index.m3u8`) don't seem to exist anymore. The reason I didn't add them to the OpenAPI specification was due to the fact that I didn't find them in the code, just two places where they're used for the exports.

**Reply by NickM-27** [reply, marked as answer]
The endpoints do exist, they are provided by nginx. 

The problem in your examples is you have /api/ but that is not in the docs I linked.

**Reply by iursevla** [reply]
Ah got it, you're correct, thanks. Now I can get the data.

Isn't that something that should be updated so that all endpoints are behind the same prefix?

**Reply by NickM-27** [reply]
No, because as I said it is provided by nginx. Nginx has /vod/ and then nginx calls the frigate api. Nginx is handling the creation of the m3u8 playlists
